### PR TITLE
Edit patient command and parser

### PR DESCRIPTION
**Pull request by owen-yap**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditPatientCommand.java [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+        this.index = index;
+        this.editPatientDescriptor = new EditPatientDescriptor(editPatientDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Patient> lastShownList = model.getFilteredPatientList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
```
Comment (praveenkrishna0512) [inline src/main/java/seedu/address/logic/commands/EditPatientCommand.java:87]: Do we want to check if Index is below 0 too?

File: src/main/java/seedu/address/logic/commands/EditPatientCommand.java
Comment (praveenkrishna0512) [inline src/main/java/seedu/address/logic/commands/EditPatientCommand.java:1]: Logic flow of `EditPatientCommand `makes sense!

**Review by praveenkrishna0512 — COMMENTED** [review]
LGTM! Just one area to apply defensive programming

File: src/main/java/seedu/address/logic/commands/EditPatientCommand.java (hunk 1 above)
Comment (owen-yap) [reply src/main/java/seedu/address/logic/commands/EditPatientCommand.java:87]: This conditional should handle that scenario

**Review by praveenkrishna0512 — APPROVED** [review]
LGTM!

**Pull request merged**